src/main/java/Duke.java

```diff
@@ -1,10 +1,146 @@
-public class Duke {
+import java.util.Scanner;
+
+class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    public void markAsNotDone() {
+        isDone = false;
+    }
+
+    @Override
+    public String toString() {
+        return "[T][" + (isDone ? "X" : " ") + "] " + description;
+    }
+}
+
+class Deadline extends Task {
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D][" + (isDone ? "X" : " ") + "] " + description + " (by: " + by + ")";
+    }
+}
+
+class Event extends Task {
+    protected String start;
+    protected String end;
+
+    public Event(String description, String start, String end) {
+        super(description);
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    public String toString() {
+        return "[E][" + (isDone ? "X" : " ") + "] " + description + " (from: " + start + " to: " + end + ")";
+    }
+}
+
+class DukeException extends Exception {
+    public DukeException(String message) {
+        super(message);
+    }
+}
+
+public class MyChatBot {
     public static void main(String[] args) {
```

Comment:
You could consider refactoring this main method to avoid long methods and adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)